presto-main/src/main/java/com/facebook/presto/sql/planner/ExpressionInterpreter.java

```diff
@@ -936,7 +940,28 @@ public Object visitCast(Cast node, Object context)
         @Override
         protected Object visitArrayConstructor(ArrayConstructor node, Object context)
         {
-            return visitFunctionCall(new FunctionCall(QualifiedName.of(ArrayConstructor.ARRAY_CONSTRUCTOR), node.getValues()), context);
+            Type elementType = ((ArrayType) expressionTypes.get(node)).getElementType();
+            List<Object> arrayValues = new ArrayList<>();
+            boolean containsExpression = false;
+            for (Expression expression : node.getValues()) {
+                Object value = process(expression, context);
+                if (value instanceof Expression) {
+                    containsExpression = true;
+                    break;
+                }
+                arrayValues.add(value);
+            }
+
+            if (containsExpression) {
+                return visitFunctionCall(new FunctionCall(QualifiedName.of(ArrayConstructor.ARRAY_CONSTRUCTOR), node.getValues()), context);
```

Comment:
i think you can just inline this return statement where you put the break statement. then you don't need the extra boolean.